### PR TITLE
Rolling file by date writes to original file

### PR DESCRIPTION
**Pull request by jonsyu1**
This adds a .seelog_rolling_tracker to track the timestamp of the last log roll.

### CODE REVIEW

File: writers_rollingfilewriter.go [hunk 1]
```diff
@@ -449,7 +461,12 @@ func (rw *rollingFileWriter) Write(bytes []byte) (n int, err error) {
 	// needs to roll if:
 	//   * file roller max file size exceeded OR
 	//   * time roller interval passed
-	nr, err := rw.self.needsToRoll()
+	fi, err := os.Stat(rw.rollingTrackerPath())
+	if err != nil {
+		return 0, err
+	}
+	lastRollTime := fi.ModTime()
```
Comment (nmiyake) [inline writers_rollingfilewriter.go:432]: Assumption seems to be that last time file was modified was when it was "rolled" -- is this accurate? Would expect last mod time was last time log event was written

File: writers_rollingfilewriter.go [hunk 2]
```diff
@@ -698,28 +710,12 @@ func (rwt *rollingFileWriterTime) sortFileRollNamesAsc(fs []string) ([]string, e
 	return ss.data, nil
 }
 
-func (rwt *rollingFileWriterTime) getNewHistoryRollFileName(lastRollName string) string {
-	return ""
+func (rwt *rollingFileWriterTime) getNewHistoryRollFileName(lastRoll rollInfo) string {
+	return lastRoll.Time.Format(rwt.timePattern)
```
Comment (nmiyake) [inline writers_rollingfilewriter.go:660]: Name is just the time?

File: writers_rollingfilewriter.go [hunk 3]
```diff
@@ -698,28 +710,12 @@ func (rwt *rollingFileWriterTime) sortFileRollNamesAsc(fs []string) ([]string, e
...
-	}
-
-	switch rwt.nameMode {
-	case rollingNameModePostfix:
-		return originalFileName + rollingLogHistoryDelimiter + time.Now().Format(rwt.timePattern), nil
-	case rollingNameModePrefix:
-		return time.Now().Format(rwt.timePattern) + rollingLogHistoryDelimiter + originalFileName, nil
-	}
-	return "", fmt.Errorf("Unknown rolling writer mode. Either postfix or prefix must be used")
+	return originalFileName, nil
```
Comment (nmiyake) [inline writers_rollingfilewriter.go]: Is this function required by the interface? If not, seems simple enough to inline

File: writers_rollingfilewriter.go (hunk 1 above)
Comment (jonsyu1) [inline writers_rollingfilewriter.go:432]: Note the difference between `rw.rollingTrackerPath()` and `rw.originalFileName`: the tracker path is only touched in `rw.createFileAndFolderIfNeeded`, which is only called by `rw.Write` on initialization or on roll. I originally named this `lastRollPath` to make that more clear, but it was easily confused with the filepath of the last roll. Any better suggestions?

File: writers_rollingfilewriter.go (hunk 2 above)
Comment (jonsyu1) [inline writers_rollingfilewriter.go:660]: RollFileName is the prefix/suffix for the rolled file. I had a similar reaction and tried to jam the `originalFileName` into this. That's how I ended up with `axiom-console.log.axiom-console.log-2016-05-19`

File: writers_rollingfilewriter.go (hunk 3 above)
Comment (jonsyu1) [inline writers_rollingfilewriter.go]: Yes. The reason this function even existed on the interface was the way roll by time behaved. This is what I described here: https://github.com/cihub/seelog/pull/117#issuecomment-220487537

**Comment by nmiyake** [comment]
Looks good

**Pull request merged**